File: an2_sem4/MPP/Homework/sapt10/cs/FestivalSellpoint/FestivalSellpoint.Network/ObjectProtocol/SpectacoleResponse.cs
using FestivalSellpoint.Domain;
using FestivalSellpoint.Network.DTO;
using System;
using System.Linq;

namespace FestivalSellpoint.Network.ObjectProtocol
{
    [Serializable]
    public class SpectacoleResponse : IResponse
    {
        public SpectacolDTO[] Spectacole { get; }

        public SpectacoleResponse(Spectacol[] spectacole)
            => Spectacole = spectacole.Select(SpectacolDTO.FromSpectacol).ToArray();
    }
}

File: an2_sem4/MPP/Homework/sapt14/network/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/network/objectprotocol/SpectacoleResponse.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.objectprotocol;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.dto.DTOFactory;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.dto.SpectacolDTO;

import java.util.Arrays;
import java.util.List;

public class SpectacoleResponse implements IResponse {
    private final SpectacolDTO[] spectacole;

    protected SpectacoleResponse(Spectacol[] spectacole){
        this.spectacole = DTOFactory.getDTO(Arrays.asList(spectacole));
    }

    public List<SpectacolDTO> getSpectacole() {
        return Arrays.asList(spectacole);
    }
}
